Blog post about the 2026 Cook-off (#584)
ProjectPythia/projectpythia.github.io: portal/posts/2026/cookoff-2026.md

```{figure} images/cookoff2026-group.jpg
:alt: Pythia Cook-off 2026 group photo on the steps in the Mesa Lab atrium

Photo credit: Samantha Scalice (with [Juan Diego Mantilla](https://github.com/jdmantillaq)'s nice camera!)
```

## More cooks, more Cookbooks

### Community growth

One of the main goals of the Pythia Cook-off is to disseminate skills and enthusiasm for open knowledge-sharing in the space of computational geoscience workflows. This year, we welcomed about 50 participants with a good mix of science backgrounds, career stages, and prior hacking experience.

There are now 153 total members of the [ProjectPythia GitHub organization](https://github.com/orgs/ProjectPythia/people), many of whom have participated in one or more of our hackathons over the past few years.

:::::{figure}
:alt: Photos of breakout groups at work

::::{grid} 3 3 3 3
:::{grid-item}
![](images/hacker1.jpg)
:::
:::{grid-item}
![](images/hacker2.jpg)
:::
:::{grid-item}
![](images/hacker3.jpg)
:::
:::{grid-item}
![](images/hacker4.jpg)
:::
:::{grid-item}
![](images/hacker5.jpg)
:::
:::{grid-item}
![](images/hacker6.jpg)
:::
::::

Photo credits: Juan Diego Mantilla
:::::

> ### Expanding the Cookbook gallery

Reviewer: Could we cross-link the other blog post I posted about the listing plugin improvements etc? That could be a nice tie-in to the infrastructure-focused impact goals

- [Spectral analysis for geophysical date](https://github.com/ProjectPythia/spectral-analysis-cookbook) with group leader [Juan Diego Mantilla](https://github.com/jdmantillaq)
- [NSF NCAR's Geoscience Data Exchange (GDEX)](https://github.com/ProjectPythia/gdex-cookbook) with group leader [Harsha Hampapura](https://github.com/hrhampapura)
- Pythia core infrastructure with group leader [Chris Holdgraf](https://github.com/choldgraf)

:::::{figure}
:alt: Photos of breakout groups at work

::::{grid} 3 3 3 3
:::{grid-item}
![](images/breakout1.jpeg)
:::

:::{grid-item}
![](images/breakout2.jpeg)
:::

:::{grid-item}
![](images/breakout3.jpeg)
:::
::::
:::::

About half of our group leaders are returning Cook-off participants, while the other half are brand-new contributors to Project Pythia! We are grateful to the whole group for bringing so many ideas, enthusiasm, and leadership to their groups.

Leaders will be working remotely with their groups over the coming weeks to put some finishing touches on their Cookbook projects before they appear in the [gallery](https://cookbooks.projectpythia.org). Watch this space!

## Challenges and solutions

### Infrastructure

> *Shout out to 2i2c and Angus. Discuss challenges around provisioning computational resources, collaboration through GitHub, etc.*

Reviewer: Would be great to do a little "lessons learned" debrief in the retro and use that to seed this section

---
date: 2026-06-26
author: brian-rose
title: Another great week of hacking on Cookbooks
subtitle: Celebrating the 2026 Pythia Cook-off hackathon at NSF NCAR
description: Celebrating the 2026 Pythia Cook-off hackathon at NSF NCAR, June 15-18 2026.
tags: ["cook-off", "cookbooks"]
---

> ## Another Pythia hackathon: cooked, served, and shared

Reviewer: Maybe promote this to the title instead of having both a title and starting the post with a section header? not a big deal tho, I like the cooked, served, and shared line 😅

Author: I like this! Adopted in the latest commit.

There are now 153 total members of the [ProjectPythia GitHub organization](https://github.com/orgs/ProjectPythia/people), many of whom have participated in one or more of our hackathons over the past few years.

:::::{figure}
:alt: Photos of breakout groups at work

::::{grid} 3 3 3 3
:::{grid-item}
![](images/hacker1.jpg)
:::
:::{grid-item}
![](images/hacker2.jpg)
:::
:::{grid-item}
![](images/hacker3.jpg)
:::
:::{grid-item}
![](images/hacker4.jpg)
:::
:::{grid-item}
![](images/hacker5.jpg)
:::
:::{grid-item}
![](images/hacker6.jpg)
:::
::::

Photo credits: Juan Diego Mantilla
:::::

### Expanding the Cookbook gallery

The other main goal of our Cook-offs is to build cool and useful new content for the [Pythia Cookbook Gallery](https://cookbooks.projectpythia.org).

Each year we welcome proposals for new Cookbooks. People bring ideas (and often some prexisting code) from their own work, and pitch their Cookbook projects to the whole community. Breakout groups are formed under the guidance of the individual topic leaders.

This year we had eight breakout groups including seven Cookbook-focussed groups:

> - [Meteorological feature tracking](https://github.com/ProjectPythia/feature-tracking-cookbook) with group leader [Matthew Lynne](https://github.com/mattsl21)

Reviewer: What if we inserted our own little "listing" here? I think you could do this with a little YAML file, then you'd be able to show the `.jpegs` of each cookbook? Like a little mini cookbook gallery? If you're interested in that lemme know and maybe I can hack together a quick improvement to myst-listing to make that work

Author: I mused a bit on how much attention to draw to the individual cookbooks. Because the various groups are at different stages of readiness for public eyes on the work, I settled on having more of a listing of breakout group topics rather than a gallery of cookbooks.

My hope is that we'll get a good fraction of these into the Pythia gallery by end of summer (many are basically ready for it now), and then do another blog post [kind of like this one](https://projectpythia.org/posts/2025/new-cookbooks/) that shouts out the content more clearly.

- [NSF NCAR's Geoscience Data Exchange (GDEX)](https://github.com/ProjectPythia/gdex-cookbook) with group leader [Harsha Hampapura](https://github.com/hrhampapura)
- Pythia core infrastructure with group leader [Chris Holdgraf](https://github.com/choldgraf) — see [Chris's blog post](gallery-listing.md)

:::::{figure}
:alt: Photos of breakout groups at work

::::{grid} 3 3 3 3
:::{grid-item}
![](images/breakout1.jpeg)
:::

:::{grid-item}
![](images/breakout2.jpeg)
:::

:::{grid-item}
![](images/breakout3.jpeg)
:::
::::

Breakout groups hard at work! Photo credits: John Clyne.
:::::

About half of our group leaders are returning Cook-off participants, while the other half are brand-new contributors to Project Pythia! We are grateful to the whole group for bringing so many ideas, enthusiasm, and leadership to their groups.

Leaders will be working remotely with their groups over the coming weeks to put some finishing touches on their Cookbook projects before they appear in the [gallery](https://cookbooks.projectpythia.org). Watch this space!

## Challenges and solutions

### Infrastructure

> We want to get our heterogenous breakout groups up and running on Cookbook development as quickly as possible on day 1 of the event, with a minimum of configuration. A cloud-based JupyterHub provisioned by our partners at [2i2c](https://2i2c.org) meant not losing any time to managing Python environments or struggling with data access on individual laptops.

Reviewer: ```suggestion
We want to get our heterogenous breakout groups up and running on Cookbook development as quickly as possible on day 1 of the event, with a minimum of configuration. A cloud-based JupyterHub provisioned by our partners at [2i2c](https://2i2c.org) meant not losing any time to managing Python environments or struggling with data access on individual laptops. You can [find our hub configuration here](https://github.com/2i2c-org/infrastructure/blob/189e605be1251d95e8280c64750c873a0f55489c/config/clusters/projectpythia/common.values.yaml).
```

> Because every Cookbook defines a bespoke computational environment with a unique set of packages and dependencies, we could not use a single shared for every group. Instead, 2i2c provided a flexible tool based on [repo2docker](https://repo2docker.readthedocs.io/en/latest/) for building and deploying a unique hub environment for each breakout group. Groups could then tailor their environemnt to the needs of their Cookbook but still work in a common shared space.

Reviewer: ```suggestion
Because every Cookbook defines a bespoke computational environment with a unique set of packages and dependencies, we could not use a single shared environment for every group. Instead, 2i2c configured the Pythia hub to use [JupyterHub Fancy Profiles](https://2i2c.org/jupyterhub-fancy-profiles/) with on-demand user image building enabled (based on [repo2docker](https://repo2docker.readthedocs.io/en/latest/)). This allowed each breakout group to build and deploy their own unique hub environment. Groups could then tailor their environment to the needs of their Cookbook but still work in a common shared space.
```

Special thanks and gratitude go to [Angus Hollands](https://2i2c.org/author/angus-hollands/) for working through some day 1 technical glitches in real time from a very different timezone!

### Breakout group logistics

> Breakout groups benefit from two different types of leadership: the _disciplinary knowledge_, vision, and mentorship skills necessary to organize the team and formulate good content, and technical navigation through the challenges of git, GitHub, and the Cookbook infrastructure.

Reviewer: ```suggestion
Breakout groups benefit from two different types of leadership: the _disciplinary knowledge_, vision, and mentorship skills necessary to organize the team and formulate good content, as well as technical navigation through the challenges of git, GitHub, and the Cookbook infrastructure.
```

Reviewer: I find this makes these multi-part sentences easier to parse haha

:::{figure} ./images/hacker7.jpeg

Group collaboration with embedded technical support. Photo credit: John Clyne.
:::

The qualitative impression of the hackathon organizers is that this arrangement worked pretty well. Most groups were able to stay focussed on content creation for a good fraction of the week. Some survey comments bear this out:

> The organizers were very prepared and the event ran smoothly. We could focus predominantly on hacking and spent little energy on spin-up.

> I liked the amount of support and guidance from the main team. I found it very helpful. I also liked all the opportunities to get to know others.

> Mainly, I appreciated the organization and the level of teamwork among the breakout groups.

> I especially appreciated the "live tech support."

> From my perspective, getting a bunch of strangers to be able to work together in a short period of time is miraculous and arguably, one of the key benefits of this hackathon.

### Thanks to our tech helpers!

A huge thanks to our volunteer team of technical facilitators:
- Robert Ford
- Ana Espinoza
- Drew Camron
- Hasnat Aslam
- Daniel Howard
- James Munroe
- Orhan Eroglu
- Rudy Klucik

> - Brian Rose

Reviewer: ```suggestion
- Brian Rose
- Chris Holdgraf
```

Is this OK? I think I was technically this role though I ended up getting sucked into high-level cloud tech facilitation land as well...no worries if you choose not to add me to this list!

## Cook-off 2026 by the numbers

- Registered participants: 49
- Travel awards: 27
- First timers: 34
- Breakout groups: 8
- Technical facilitators: 11

> - Total commits to repositories in the [ProjectPythia GitHub organization](https://github.com/ProjectPythia): 842

Reviewer: wow!

Author: Yep that's a fun statistic. I needed some help from Claude to write script to count these using the gh api.

- Fraction of surveyed participants who indicated they would "definitely" or "probably" participate in a similar event in the future: 90%

## The Albany gang

Finally, a special shout-out to the gang from the University at Albany's [Department of Atmospheric & Environmental Sciences](https://www.albany.edu/daes) who were well-represented at this year's Cook-off!

```{figure} images/Pythia-2026-UAlbany.jpeg
:alt: Pythia Cook-off 2026 participants from the University at Albany in front of the Mesa Lab main entrance

Cook-off 2026 participants from [UAlbany DAES](https://www.albany.edu/daes). From left to right: Kevin Tyle, Juan Diego Mantilla, Kathryn Rooney, Bella Condo, Brian Rose, Robert Ford, Matthew Lynne, Alex Blackmer, Jacob Vile.
```

> UAlbany is a lead partner with NSF NCAR on Project Pythia. This year we had a record-breaking nine participants travel from Albany to Boulder for the event, including three breakout group leaders and two technical facilitators. Go [Great Danes](https://en.wikipedia.org/wiki/Albany_Great_Danes)!

Reviewer: TIL that the Great Dane is a mascot :-D

## Challenges and solutions

### Infrastructure

We want to get our heterogenous breakout groups up and running on Cookbook development as quickly as possible on day 1 of the event, with a minimum of configuration. A cloud-based JupyterHub provisioned by our partners at [2i2c](https://2i2c.org) meant not losing any time to managing Python environments or struggling with data access on individual laptops.

Because every Cookbook defines a bespoke computational environment with a unique set of packages and dependencies, we could not use a single shared for every group. Instead, 2i2c provided a flexible tool based on [repo2docker](https://repo2docker.readthedocs.io/en/latest/) for building and deploying a unique hub environment for each breakout group. Groups could then tailor their environemnt to the needs of their Cookbook but still work in a common shared space.

Special thanks and gratitude go to [Angus Hollands](https://2i2c.org/author/angus-hollands/) for working through some day 1 technical glitches in real time from a very different timezone!

### Breakout group logistics

> *Talk about separating content leadership from technical facilitation, and embedding a tech facilitator in each group. Shout out all facilitators*

Reviewer 2: ```suggestion
```

> From my perspective, getting a bunch of strangers to be able to work together in a short period of time is miraculous and arguably, one of the key benefits of this hackathon.

### Thanks to our tech helpers!

A huge thanks to our volunteer team of technical facilitators:
- Robert Ford
- Ana Espinoza
- Drew Camron
- Hasnat Aslam
- Daniel Howard
- James Munroe
- Orhan Eroglu
- Rudy Klucik
- Brian Rose

## Cook-off 2026 by the numbers

- Registered participants: 49
- Travel awards: 27
- First timers: 34
- Breakout groups: 8
- Technical facilitators: 11
- Total commits to repositories in the [ProjectPythia GitHub organization](https://github.com/ProjectPythia): 842
- Fraction of surveyed participants who indicated they would "definitely" or "probably" participate in a similar event in the future: 90%

## The Albany gang

Finally, a special shout-out to the gang from the University at Albany's [Department of Atmospheric & Environmental Sciences](https://www.albany.edu/daes) who were well-represented at this year's Cook-off!

```{figure} images/Pythia-2026-UAlbany.jpeg
:alt: Pythia Cook-off 2026 participants from the University at Albany in front of the Mesa Lab main entrance

> Cook-off 2026 participants from [UAlbany DAES](https://www.albany.edu/daes). From left to right: Kevin Tyle, Juan Diego Mantilla, Kathryn Rooney, Bella Condo, Brian Rose, Robert Ford, Matthew Lynne, Alex Blackmer, Jacob Vile.

Reviewer 2: For consistency, do we know the photo credit?

Author: I don't! I think it was whoever Bella handed her phone to?
